File: week02/Jisoo/02_같은_숫자는_싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        
        
        List<Integer> list = new ArrayList<>();
        for(int num : arr){
            if(list.size()==0){
                list.add(num);
            } else if(list.getLast() != num) {
                list.add(num);
            }
        }
        
        int[] answer = new int[list.size()];
        for(int i=0; i<list.size(); i++){
            answer[i] = list.get(i);
        }
        return answer;
    }
}

File: week02/sujin/02_같은_숫자는_싫어.py
def solution(arr):
    n = len(arr)
    num = arr[0]
    answer = [num]
    
    for i in range(1, n):
        if arr[i] == num:
            continue
            
        num = arr[i]
        answer.append(num)
    
    
    return answer